packages/ui/src/RangeBar.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  ForwardedRef,
+  forwardRef,
+  HTMLProps,
+} from 'react'
+import styled from 'styled-components'
+
+import { Text } from './Text'
+
+export interface RangeBarProps extends HTMLProps<HTMLInputElement> {
+  /**
+   * Step value of scroll bar
+   */
+  stepvalue?: number
+}
+
+/**
+ * Height of the scroll bar wrapper container
+ */
+const containerHeight = 70
+
+export const RangeBar = forwardRef(
+  (
+    {
+      type = 'range',
+      className = '',
+      disabled = false,
+      stepvalue,
+      min,
+      max,
+      ...props
+    }: RangeBarProps,
+    ref: ForwardedRef<HTMLInputElement>,
+  ) => {
+    const [valuePercent, setValuePercent] = useState(0)
+    const [rangeValue, setRangeValue] = useState(0)
+
+    const calculatePercent = (val: string) => {
+      setValuePercent((parseInt(val, 10) / Number(max)) * 100)
+    }
+
+    const handleRangeChange = e => {
+      if (e && e.target && e.target.value) {
+        setRangeValue(e.target.value)
+        calculatePercent(e.target.value)
+      }
+    }
+
+    useEffect(() => {
+      const startValue = (Number(max) - Number(min)) / 2
+      setRangeValue(startValue)
+      calculatePercent(startValue.toString())
+    }, [])
+
+    useEffect(() => {
+      const startValue = (Number(max) - Number(min)) / 2
+      setRangeValue(startValue)
+      calculatePercent(startValue.toString())
+    }, [min, max])
+
+    return (
+      <Wrapper>
+        {props.title && (
+          <Title color="oxfordBlue" type="body 2">
+            {props.title}
+          </Title>
+        )}
+        <InnerWrapper className={className} valuePercent={valuePercent}>
+          <input
+            ref={ref}
+            value={rangeValue}
+            type={type}
+            min={min}
+            step={stepvalue}
+            max={max}
+            onChange={handleRangeChange}
+            {...props}
+            disabled={disabled}
+          />
+        </InnerWrapper>
+      </Wrapper>
+    )
+  },
+)
+
+const Wrapper = styled.div`
+  height: ${containerHeight}px;
+  width: 100%;
+
+  #bottom-label {
+    transform: translate(-10px, -10px);
+  }
+`
+/* stylelint-disable property-no-vendor-prefix */
+const InnerWrapper = styled.div<{ valuePercent: number }>`
+  input {
+    appearance: none;
+    outline: 0;
+    height: 15px;
+    border-radius: 4px;
+    background: ${({ theme, valuePercent }) =>
+      `linear-gradient(to right, ${theme.buttonDark} 0%, ${theme.buttonDark} ${
+        valuePercent - 1
+      }%, #DAE3ED ${valuePercent - 1}%, #DAE3ED 100%);`};
+    box-shadow: 0 0 4px rgb(0 0 0 / 0.5);
```

Comment:
by design there is no box-shadow here
![image](https://user-images.githubusercontent.com/22115870/188407426-4b9bf3cf-35d0-45d8-87ac-365e3dbddd47.png)
![image](https://user-images.githubusercontent.com/22115870/188407600-a67a9512-cda3-43d0-9e3f-4b9e80409637.png)

only on "thumb"

Reply:
It would be benefitial, not to go back to this component.
meaning that in a storybook, we should have a complete example of how this should look.

![image](https://user-images.githubusercontent.com/22115870/188407854-8628fb19-09ac-4d8a-81f5-9897f18f2665.png)

meaning that `min={1} max={40} step={12}` should look like in design.
with below labels "1 month,etc."

Reply:
Done like static text, but we need to resolve labels with PO and designers